plugins/node/opentelemetry-instrumentation-mysql2/src/instrumentation.ts

```diff
@@ -210,4 +211,17 @@ export class MySQL2Instrumentation extends InstrumentationBase<MySQL2Instrumenta
       };
     };
   }
+
+  private _getConnectionPrototype(moduleExports: any): mysqlTypes.Connection {
+    const baseClass = Object.getPrototypeOf(moduleExports.Connection);
+
+    // if `baseClass` has no prototype means connection is not extending
+    // another class so the functions we need to patch are already there
+    if (typeof baseClass.prototype === 'undefined') {
```

Comment:
Ah, I think in older versions you get `EventEmitter` as a base. That may be why the test are failing.

Reply:
I've reverted the check to make it more explicit. It will return the base class prototype if it has the `query` function (`execute` goes together). Otherwise it falls back to the `Connection` prototype.

Note: I had trouble running it locally since docker image `mysql:5.7` has no support for `linux/arm64/v8` platform. Is it possible to update the image version in `@opentelemetry/contrib-test-utils`?